### PR TITLE
model_viewer: Allow viewing/selecting objects within a HOB

### PR DESCRIPTION
**Pull request by BenChampion**
I thought it would useful to view all objects in a HOB to help identify the unknown HOB fields (e.g. we can see which objects' textures look OK and which ones don't).

Note:
- It seems to work but this is my first time working with Pascal, so please check I've not done anything crazy!
- Obj export only does first object in file (object selection still a TODO)
- Vertices are still shown for all objects in the file (object selection still a TODO)

### CODE REVIEW

**Comment by dpethes** [comment]
Sure, good idea. Don't worry about the obj export and vertices, that can be worked on later. The naming scheme is bit of a mess, object/object part should be probably used instead of triangles/facegroups. But let's merge this first and deal with it afterwards.

File: model_viewer/hob_mesh.pas [hunk 1]
```diff
@@ -351,32 +360,37 @@ procedure TModel.DrawGL(var opts: TRenderOpts);
...
-
-  if opts.fg_all then begin
-      for k := 0 to Length(_triangles) - 1 do begin
-          if _triangles[k].Size > 0 then
-              for i := 0 to _triangles[k].Size - 1 do
-                  DrawTri(_triangles[k][i]);
+  if _triangles.Size > 0 then begin
+    j := min(integer(opts.obj_to_draw), integer(_triangles.Size - 1));
+    opts.obj_to_draw := j; // clip
+    if _triangles[j].Size > 0 then begin
```
Comment (dpethes) [inline model_viewer/hob_mesh.pas]:
```
  if _triangles[j].Size = 0 then
      exit;     
``` 
to eliminate indent

File: model_viewer/hob_mesh.pas [hunk 2]
```diff
@@ -351,32 +360,37 @@ procedure TModel.DrawGL(var opts: TRenderOpts);
   end;
 
   glColor3f(1, 1, 1);
-
-  if opts.fg_all then begin
-      for k := 0 to Length(_triangles) - 1 do begin
-          if _triangles[k].Size > 0 then
-              for i := 0 to _triangles[k].Size - 1 do
-                  DrawTri(_triangles[k][i]);
+  if _triangles.Size > 0 then begin
```
Comment (dpethes) [inline model_viewer/hob_mesh.pas]: Use ```  if _triangles.Size = 0 then
      exit; ``` at the start of the function. Does this actually happen with any file?

Comment (BenChampion) [reply model_viewer/hob_mesh.pas]: Yes, this happens with the opkg2_HOB files per level.

**Comment by BenChampion** [comment]
Thanks for the comments!

**Comment by dpethes** [comment]
Thanks for your contribution.

**Pull request merged**